packages/component-model-manuscript/entities/manuscript/index.js

```diff
@@ -189,64 +183,6 @@ class Manuscript extends BaseModel {
     await this.$loadRelated('[teams, files]')
   }
 
-  async needsRefresh(trx = null) {
-    const current = await this.constructor.query(trx).findById(this.id)
-
-    const storedUpdateTime = new Date(current.updated).getTime()
-    const instanceUpdateTime = new Date(this.updated).getTime()
-
-    const result = instanceUpdateTime < storedUpdateTime
-    if (result) {
-      logger.info(
-        `\n\n\nneeds refresh ${storedUpdateTime} : ${instanceUpdateTime}`,
-      )
-    }
-    return result
-  }
-
-  async save(options = {}) {
```

Comment:
I don't think I agree with replacing the save method with saveGraph. Looking at database logs, uploading files updates the manuscript table and that will likely result in the occasional race condition.

Reply:
Not sure I follow? Your existing method used `upsertGraphAndFetch` https://github.com/elifesciences/elife-xpub/pull/2131/files#diff-15ad46ae309798af5650212060f56fd3L211

The difference is that you were previously refetching during a transaction in certain cases, instead of using a `forUpdate` lock on the rows that are being updated.

I do see the potential issue with graph inserts affecting nested relationships, which won't be locked. But the way to solve that is a `forUpdate` on the `upsertGraph`, locking all the affected rows and not just the parent, like described here: https://github.com/Vincit/objection.js/issues/1127#issuecomment-433819227

I'll get to implementing that over in PubSweet's side right away, seems like it has no downsides.

P.S.: Out of curiosity, what gets updated on the Manuscript when the files change?

Reply:
The last time I explained this there was a full sequence diagram :laughing: on a white board ... let's have a call sometime @jure